src/gui/advancedsettings.cpp

```diff
@@ -333,12 +332,41 @@ void AdvancedSettings::updateCacheSpinSuffix(const int value)
 }
 #endif
 
-void AdvancedSettings::updateSaveResumeDataIntervalSuffix(const int value)
+#ifdef QBT_USES_CUSTOMDBUSNOTIFICATIONS
+void AdvancedSettings::updateNotificationTimeoutSuffix(const int value)
 {
-    if (value > 0)
-        m_spinBoxSaveResumeDataInterval.setSuffix(tr(" min", " minutes"));
+    if (value == 0)
+        m_spinBoxNotificationTimeout.setSuffix(tr(" (infinite)"));
+    else if (value < 0)
+        m_spinBoxNotificationTimeout.setSuffix(tr(" (system default)"));
     else
-        m_spinBoxSaveResumeDataInterval.setSuffix(tr(" (disabled)"));
+        m_spinBoxNotificationTimeout.setSuffix(tr(" ms", " milliseconds"));
+}
+#endif
+
+void AdvancedSettings::updateOutgoingPortsSuffix(QSpinBox &spinbox)
+{
+    spinbox.setSuffix((spinbox.value() > 0) ? QString() : tr(" (disabled)"));
```

Comment:
In cases where there is only one special value (and this is the vast majority of cases), it would still be easier to use "special value text" property instead of handling each change manually and modify the suffix:
```c++
spinbox.setSpecialValueText("0 (disabled)");
```

Reply:
Done.